akka-persistence/src/main/scala/akka/persistence/journal/leveldb/LeveldbStore.scala

```diff
@@ -42,9 +42,24 @@ private[persistence] trait LeveldbStore extends Actor with WriteJournalBase with
 
   def asyncWriteMessages(messages: immutable.Seq[AtomicWrite]): Future[immutable.Seq[Try[Unit]]] =
     Future.fromTry(Try {
-      withBatch(batch ⇒ messages.map { a ⇒
-        Try(a.payload.foreach(message ⇒ addToMessageBatch(message, batch)))
-      })
+      withBatch { batch ⇒
+        var result: immutable.Seq[Try[Unit]] = Nil
+        val iter = messages.iterator
+        var n = 0
+        while (iter.hasNext) {
+          try {
+            iter.next().payload.foreach(message ⇒ addToMessageBatch(message, batch))
+            n += 1
+          } catch {
+            case NonFatal(e) ⇒
+              if (result.isEmpty)
+                result = Vector.fill(n)(AsyncWriteJournal.successUnit)
+              result :+= Failure(e)
+          }
+        }
+
+        result
+      }
```

Comment:
you didn't say that it would take 2 x 36 minutes ;-)
I'll do the second run when having dinner or something.

Here is the first run, with the optimized code:

    [info] Benchmark                                                                                    Mode  Cnt       Score      Error  Units
    [info] PersistentActorThroughputBenchmark.actor_normalActor_reply_baseline                         thrpt  200  146964.946 ± 2766.057  ops/s
    [info] PersistentActorThroughputBenchmark.persistentActor_noPersist_reply                          thrpt  200  149247.984 ± 2760.938  ops/s
    [info] PersistentActorThroughputBenchmark.persistentActor_persistAsync_reply                       thrpt  200  107536.480 ± 2585.782  ops/s
    [info] PersistentActorThroughputBenchmark.persistentActor_persistAsync_replyRightOnCommandReceive  thrpt  200  110368.506 ± 2786.160  ops/s
    [info] PersistentActorThroughputBenchmark.persistentActor_persist_reply                            thrpt  200    9601.518 ±   81.373  ops/s